airflow/providers/dbt/cloud/operators/dbt.py

```diff
@@ -250,6 +250,152 @@ def get_openlineage_facets_on_complete(self, task_instance) -> OperatorLineage:
+            account_id=self.account_id,
+            payload={
+                "job_definition_id": self.job_id,
+                "status": DbtCloudJobRunStatus.NON_TERMINAL_STATUSES.value,
+                "order_by": "-created_at",
+            },
+        ).json()["data"]
+
+        if non_terminal_runs:
+            self.run_id = non_terminal_runs[0]["id"]
+            job_run_url = non_terminal_runs[0]["href"]
+            self.log.info("Non terminal runs for this job exist. Skipping retry.")
+        else:
+            retry_job_response = self.hook.retry_job_run(
+                account_id=self.account_id,
+                job_id=self.job_id,
+            )
+            self.run_id = retry_job_response.json()["data"]["id"]
+            job_run_url = retry_job_response.json()["data"]["id"]
+
+        # Push the ``job_run_url`` value to XCom regardless of what happens during execution so that the job
+        # run can be monitored via the operator link.
+        context["ti"].xcom_push(key="job_run_url", value=job_run_url)
+
+        if self.wait_for_termination and isinstance(self.run_id, int):
+            if self.deferrable is False:
+                self.log.info("Waiting for job run %s to terminate.", self.run_id)
+
+                if self.hook.wait_for_job_run_status(
+                    run_id=self.run_id,
+                    account_id=self.account_id,
+                    expected_statuses=DbtCloudJobRunStatus.SUCCESS.value,
+                    check_interval=self.check_interval,
+                    timeout=self.timeout,
+                ):
+                    self.log.info("Job run %s has completed successfully.", self.run_id)
+                else:
+                    raise DbtCloudJobRunException(f"Job run {self.run_id} has failed or has been cancelled.")
+
+                return self.run_id
+            else:
+                end_time = time.time() + self.timeout
+                job_run_info = JobRunInfo(account_id=self.account_id, run_id=self.run_id)
+                job_run_status = self.hook.get_job_run_status(**job_run_info)
+                if not DbtCloudJobRunStatus.is_terminal(job_run_status):
+                    self.defer(
+                        timeout=self.execution_timeout,
+                        trigger=DbtCloudRunJobTrigger(
+                            conn_id=self.dbt_cloud_conn_id,
+                            run_id=self.run_id,
+                            end_time=end_time,
+                            account_id=self.account_id,
+                            poll_interval=self.check_interval,
+                        ),
+                        method_name="execute_complete",
+                    )
+                elif job_run_status == DbtCloudJobRunStatus.SUCCESS.value:
+                    self.log.info("Job run %s has completed successfully.", self.run_id)
+                    return self.run_id
+                elif job_run_status in (
```

Comment:
@andyguwc is there any reason to check the cancelled and error in the same condition statement? exception raised could be more specific if we put in different conditions right?

Reply:
this logic came from the existing DBT operator. I think the rationale is we probably just care about success. And CANCELLED ERROR needs manual triaging anyways. 

That said, I see the other comment too about why creating a new operator which duplicates a lot of code. Let me think about the other approach as well.